lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -0,0 +1,147 @@
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*****************************************************************************
+  Source      	SetMessageManagerHandler.cpp
+  Version     	0.1
+  Date       	2020/08/08
+  Product     	SessionD
+  Subsystem   	5G Landing object in SessionD
+  Author/Editor Sanjay Kumar Ojha
+  Description 	Acts as 5G Landing object in SessionD & start 5G related flow
+*****************************************************************************/
+#include <chrono>
+#include <thread>
+
+#include <google/protobuf/util/time_util.h>
+
+#include "SetMessageManagerHandler.h"
+#include "magma_logging.h"
+#include "GrpcMagmaUtils.h"
+
+using grpc::Status;
+
+namespace magma
+{
+
+/**
+ * SetMessageManagerHandler processes gRPC requests for the sessionD
+ * This composites the earlier LocalSessionManagerHandlerImpl and uses the
+ * exiting functionalities.
+ */
+
+/*Constructor*/
+SetMessageManagerHandler::SetMessageManagerHandler(
+    std::shared_ptr<SessionStateEnforcer> m5genforcer, SessionStore& session_store)
+    : session_store_(session_store),
+      m5g_enforcer_(m5genforcer) {}
+
+/*Building session config with required parameters*/  //TODO on SubscriberID how to assigne or manage
+SessionConfig SetMessageManagerHandler::m5g_build_session_config(
+              const SetSMSessionContext& request) {
+  SessionConfig cfg ;
+  /*copying pnly 5G specific data to respective elements*/
+  cfg.common_context         = request.common_context();
+  cfg.rat_specific_context   = request.rat_specific_context();
+
+  return cfg;
+}
+
+/* Handling set message from AMF
+ * check if it is INITIAL_REQUEST or EXISTING_PDU_SESSION
+ * if it is INITIAL_REQUEST need to create the session context in SessionMap
+ * and write to memory by SessionStore
+ * If EXISTING_PDU_SESSION, check on incoming session state and version
+ * accordingly take action.
+ * As per ASYNC method, response_callback is included but not functional
+ * */
+
+void SetMessageManagerHandler::SetAmfSessionContext(
+		ServerContext* context,
+		const SetSMSessionContext* request,
+		std::function<void(Status, SmContextVoid)> response_callback)
+{
+   auto& request_cpy = *request;
+   //Print the message from AMF
+   PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request_cpy));
+   response_callback(Status::OK, SmContextVoid());
+   /*The Event Based main_thread invocation and runs to handle session state*/ 
+   m5g_enforcer_->get_event_base().runInEventBaseThread (
+      [this, context, response_callback, request_cpy]() {

```

Comment:
Does context need to be captured here?

Reply:
This is ServerContext* context and leveraged from LocalSessionManagerHandler object. I believe, its related to gRPC channel. Can you pls. elaborate your intent?

Reply:
It is captured in the lambda but does not seem to be used anywhere.

Reply:
Yes, removed.